src/openlifu/vf/virtual_fit.py

```diff
@@ -0,0 +1,161 @@
+
+    volume: xa.Dataset = field(default_factory=xa.Dataset)
+    """The MRI volume on which to optimize the position."""
+
+    transducer: Transducer = field(default_factory=Transducer)
+    """Transducer that sits on the skin."""
+
+    def __post_init__(self):
+        self.logger = logging.getLogger(__name__)
+        """The VirtualFit logger."""
+        self.logger.info(f"Initializing VirtualFit with the following parameters: {self.__dict__}")
+        self.logger.info("VirtualFit: Skin extraction...")
+        # 1. extract skin surface, this is done only once at initialization
+        # self.skin_surface = self.extract_skin_surface(volume: xa.Dataset)
+        """A list of vertices representing the skin surface."""
+
+    def extract_skin_surface(self, volume: xa.Dataset, quantile: float = 0.05):
+        #TODO: basic thresholding + convex hull
+        # from scipy.spatial import ConvexHull
+        # threshold = np.quantile(volume, 0.05)   #TODO: check otsu threhsolding instead
+        # volume_thresholded = volume[volume > threshold]
+        #
+        # return ConvexHull(volume)
+        pass
+
+    def fit_to_surface(
+            self,
+            sph_coords: Tuple[float, float],
+            skin_surface: np.ndarray
+        ) -> np.ndarray:
+        """
+        Fit a 3D plane plane given spherical coordinates (yaw, pitch)
+        and a set of points coordinates (lat, ele, ax).
+        """
+        pass
+
+    def get_search_grid(
+            self,
+            yaw_range: Tuple[int, int],
+            yaw_step: int,
+            pitch_range: Tuple[int, int],
+            pitch_step: int
+        ) -> np.ndarray:
+        """
+        Defines the transducer search grid in (yaw, pitch) coordinates.
+        """
+        yaw_sequence = np.arange(yaw_range[0], yaw_range[-1], yaw_step)
+        pitch_sequence = np.arange(pitch_range[0], pitch_range[-1], pitch_step)
+        pitch_yaw_grid = np.meshgrid(pitch_sequence, yaw_sequence, indexing="ij")
+
+        return pitch_yaw_grid
+
+    def analyse_position(self, pos: np.ndarray, transducer: Transducer, target: Point):
+        """
+        Analyse the transducer position relative to a specific target.
+        """
+        #TODO: Compute if target is within steering limits
+        #TODO: In the future, we should implement the ray-tracing analysis given a full segmentation
+
+        # pos_analysis = 1.0
+        # target_tr_space = target2trspace(pos, target)
+        # for target_constraint in self.steering_limits:
+        #     pos = target_tr_space.get_position(
+        #         dim=target_constraint.dim,
+        #         units=target_constraint.units
+        #     )
+        #     try:
+        #         target_constraint.check_bounds(pos)
+        #     except ValueError:
+        #         pos_analysis = 0.0
+        #
+        # return pos_analysis
+
+        pass
+
+    def run(
+            self,
+            target: Point,
```

Comment:
What is assumed about the coordinates of this target?

E.g. is it in the same coordinates as the indices of the volume array?

Or is it in the lat, ele, ax coordinates?

Reply:
Target is in LPS coordinates (coordinates of the volume)

Reply:
In `Protocol.calc_solution` we assumed the target to be in same coordinates as the volume. Because Slicer resample to (lat, ele, ax), the target is also in that. Because we did not implement the resampling routine in the python lib we should also assume (lat, ele, ax) here for short term.